pkg/security/admission/admission.go

```diff
@@ -177,11 +196,13 @@ func assignSecurityContext(provider scc.SecurityContextConstraintsProvider, pod
 	}
 
 	if len(errs) > 0 {
+		// ensure psc is not mutated if there are errors
+		pod.Spec.SecurityContext = originalPSC
 		return errs
 	}
 
 	// if we've reached this code then we've generated and validated an SC for every container in the
-	// pod so let's apply what we generated
+	// pod so let's apply what we generated.  Note: the psc is already applied.
 	for i, sc := range generatedSCs {
 		pod.Spec.Containers[i].SecurityContext = sc
```

Comment:
Haven't I already done this here: https://github.com/openshift/origin/pull/5079/files#diff-eb43ba779f411e3ee6b772cceee9810dR194